### DISCUSSION TITLE
Support iteration through file deltas.

### DISCUSSION THREAD

**Discussion by turbocool3r** [GitoxideLabs/gitoxide#779] [Ideas]
### Summary 💡

I'm suggesting exposing delta chains from packs in a repository to accelerate incremental parsing of code for analysis. This should provide some API that returns either a full object or a structure wrapping a delta when odb is queried for an oid. Ideally this should support ref deltas.

### Motivation 🔦

I'm trying to write a tool that uses an incremental parser to statically analyze code in large Git repos fast. Since large repos are usually compressed into packs and downloaded as packs (IIRC) identifying chains of deltas that can be fed into a parser would likely provide a considerable speedup.

**Comment by Byron** [comment]
`gitoxide` provides facilities to decode packs as a whole and optimally, feeding a fully decoded object at a time to a function to handle it, using as many threads as specified.

If the problem to solve is a `fetch -> process received pack` workflow then indeed `gitoxide` will decompress the whole pack anyway (optimally, multi-threaded) and if receiving objects in any order is suitable for the workload a callback for that could certainly be added.

`gitoxide` currently doesn't validate packs it receives and adding verification would naturally offer an entry point for user-defined code to run.

In the meantime, decoding a pack optimally is already available in the `gix-pack` crate, so you would be able to try it already without any new APIs.
